### DISCUSSION TITLE
Running Carla in C++ API

### DISCUSSION THREAD

**Discussion by mohamad20111390** [carla-simulator/carla#6455] [General]
Hello everyone,

I'm completely new to Carla simulator , I tried to run it using Python API according to the tutorial, and it worked. However I need to use C++ API to run Carla Simulator but I could not find any tutorial for it. I tried to run the example on " https://github.com/carla-simulator/carla/tree/master/Examples/CppClient " as it is described , so after running Carla simulator using " ./CarlaUE4.sh " on its main directory, I tried  command "**make run**" on  " carla/Examples/CppClient " as I cloned it using git and then I got the following error  [so there are 2 separate directories , one for Carla simulator and the other for github cloned directory] :

**clang-6 is required, but it's not installed.**

However after checking clang version ( i.e clang --version ) the following message appears on my screen : 

""" clang version 6.0.1-svn334776-1~exp1~20190309042703.125 (branches/release_60)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin """

It seems clang is already installed but while  " **make run** " it can not be used. So, Can you please guide me how  to run Carla simulator using C++ API ? and do you know any sources/tutorial for running Carla Simulator using C++ API, since I could find them for python.

Thank you all

**Comment by Willis814** [comment]
Do you know how to run c++ API, I have so many questions like you😭

**Reply by mohamad20111390** [reply]
unfortunately I could not find any tutorial to explain it, there is only an example in git repository of carla, but it is too complicated for beginning.

**Comment by Sephiroth1118** [comment]
Hi! Do you find some solutions now? I'm also a beginner of Carla and need to run C++ API.
